trin-types/src/node_id.rs

```diff
@@ -1,24 +1,63 @@
-use serde::{Deserialize, Serialize};
-use serde_hex::{SerHex, StrictPfx};
-use std::ops::Deref;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use trin_utils::bytes::{hex_decode, hex_encode};
 
 type RawNodeId = [u8; 32];
 
 /// Discv5 NodeId
-// TODO: Wrap this type over discv5::NodeId type
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct NodeId(#[serde(with = "SerHex::<StrictPfx>")] pub RawNodeId);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NodeId(discv5::enr::NodeId);
 
-impl From<RawNodeId> for NodeId {
-    fn from(item: RawNodeId) -> Self {
-        NodeId(item)
+impl NodeId {
```

Comment:
Let's implement [Deref](https://doc.rust-lang.org/std/ops/trait.Deref.html) trait instead of explicitly implementing those methods.

Reply:
![image](https://user-images.githubusercontent.com/31669092/232813574-697db1e8-d80c-4570-8276-31390bf89377.png)
![image](https://user-images.githubusercontent.com/31669092/232813619-6eec4919-75ac-4d33-b1cd-4ea339d832ac.png)
How would that work with Rust? Sorry I must not be familiar enough with how Deref works, but that is what happens when I attempt to use this for example

Reply:
On line 30 you want to return `Self` which is the wrapped structure NodeId, so it should be something like:

```rust          Ok(NodeId(discv5::enr::NodeId::parse(result.as_slice()).expect("Could not parse decoded NodeID")))
```